manager/integration/tests/test_rwx.py

```diff
@@ -1,12 +1,373 @@
+            'resources': {
+                'requests': {
+                    'storage': '20G'
+                }
+            }
+        }
+    }
+
+    def finalizer():
+        api = k8sclient.CoreV1Api()
+        client = get_longhorn_api_client()
+        volumes = client.list_volume()
+
+        if not check_pvc_existence(api, pvc_manifest['metadata']['name'],
+                                   namespace='longhorn-system'):
+            return
+
+        volume_name = ''
+        for v in volumes:
+            if v.kubernetesStatus.pvcName == 'longhorn-nfs-provisioner':
+                volume_name = v.name
+                client.delete(v)
+                break
+        wait_for_volume_delete(client, volume_name)
+
+        core_api = get_core_api_client()
+        delete_and_wait_pvc(core_api, pvc_manifest['metadata']['name'])
+
+    request.addfinalizer(finalizer)
+
+    return pvc_manifest
+
+
+@pytest.fixture(scope="module", autouse="True")
+def nfs_storage_class(request):
+    sc_manifest = {
+        'apiVersion': 'storage.k8s.io/v1',
+        'kind': 'StorageClass',
+        'metadata': {
+            'name': NFS_STORAGECLASS_NAME
+        },
+        'provisioner': 'nfs.longhorn.io',
+        'mountOptions': [
+            'vers=4.1',
+            'noresvport'
+        ]
+    }
+
+    def finalizer():
+        delete_storage_class(NFS_STORAGECLASS_NAME)
+
+    request.addfinalizer(finalizer)
+
+    return sc_manifest
+
+
+@pytest.fixture(scope="module", autouse="True")  # NOQA
+def core_api(request):
+    """
+    Create a new CoreV1API instance.
+    Returns:
+        A new CoreV1API Instance.
+    """
+    c = Configuration()
+    c.assert_hostname = False
+    Configuration.set_default(c)
+    k8sconfig.load_incluster_config()
+    core_api = k8sclient.CoreV1Api()
+
+    cleanup_client()
+
+    return core_api
+
+
+@pytest.fixture(scope="module", autouse="True")
+def apps_api(request):
+    """
+    Create a new AppsV1API instance.
+    Returns:
+        A new AppsV1API Instance.
+    """
+    c = Configuration()
+    c.assert_hostname = False
+    Configuration.set_default(c)
+    k8sconfig.load_incluster_config()
+    apps_api = k8sclient.AppsV1Api()
+
+    return apps_api
+
+
+@pytest.fixture(scope="module", autouse="True")
+def longhorn_nfs_provisioner_setup(core_api, apps_api, nfs_pvc,
+                                   nfs_storage_class, deployment_nfs):
+    """
+    Deploy longhorn-nfs-provisioner pod and volume
+    """
+
+    create_storage_class(nfs_storage_class)
+    core_api.create_namespaced_persistent_volume_claim(
+        body=nfs_pvc, namespace='longhorn-system')
+    apps_api.create_namespaced_deployment(namespace='longhorn-system',
+                                          body=deployment_nfs)
+    wait_deployment_replica_ready(apps_api, 'longhorn-nfs-provisioner', 1,
+                                  namespace='longhorn-system')
+    # This is to make sure nfs server is started.
+    time.sleep(25)
```

Comment:
There is a probably a better way of checking for this, but this is okay for now.